### PR TITLE
Redirection to valid URL if project in a directory

### PR DESCRIPTION
**Pull request by RiteshParyali**
Redirection to valid url if project in a directory

### CODE REVIEW

**Review by ndm2 — COMMENTED** [review]
If this is going to be changed, then there should probably also be a comment that explains that the authentication middleware needs to be added _after_ the routing middleware.

And people that apply the authentication middleware in routing scopes are hopefully savvy enough to figure the limitations.

File: docs/en/index.rst [hunk 1]
```diff
@@ -65,7 +66,7 @@ define the ``AuthenticationService`` it wants to use. Add the following method y
 
         // Define where users should be redirected to when they are not authenticated
         $service->setConfig([
-            'unauthenticatedRedirect' => '/users/login',
+            'unauthenticatedRedirect' => Router::url(['controller' => 'users', 'action' => 'login']),
```
[the remark this answers is not in the document]
Comment (dereuromark) [reply docs/en/index.rst:69]: Shouldnt n array suffice and the URL is build then at runtime when needed through router::url() call?

[the remark this answers is not in the document]
Comment (ADmad) [reply docs/en/index.rst:69]: We have tried to keep the plugin to be usable standalone outside CakePHP apps too. Hence using router internally is avoided. Only cakephp/core is a hard dependency.

**Comment by RiteshParyali** [comment]
Suppose i have project set up in **https://xyz.com/cake**

I want to access a protected path  **https://xyz.com/cake/dashboard**.
Then it's redirect me to  **https://xyz.com/users /login** Which is invalid url.
After using Route::url(........) it's redirect me to a valid url **https://xyz.com/cake/users /login**

File: docs/en/index.rst [hunk 2]
```diff
@@ -34,6 +34,7 @@ imports::
     use Authentication\Middleware\AuthenticationMiddleware;
     use Cake\Http\MiddlewareQueue;
     use Psr\Http\Message\ServerRequestInterface;
+    use Cake\Routing\Router;
```
[the remark this answers is not in the document]
Comment (dereuromark) [reply docs/en/index.rst:36]: The order seems fine to me.

**Comment by dereuromark** [comment]
> We have tried to keep the plugin to be usable standalone outside CakePHP apps too. Hence using router internally is avoided. Only cakephp/core is a hard dependency.

The current docs now with this change ironically reflect the opposite though.
I think this then needs to be made clearer, and a string example to be kept IMO.

The other - way cleaner - way would be to allow for array and document both array (cake router available) and string (agnostic way).

File: docs/en/index.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (ndm2) [reply docs/en/index.rst:36]: Because you're looking at it 5 hours after it has been changed 😉

**Comment by markstory** [comment]
@dereuromark I think trying to make authentication work outside of Cake was a mistake, and has introduced more complexity for very little value. I'm considering making the plugin more 'cake friendly' for the next major.

**Comment by dereuromark** [comment]
There is 0 users outside of cake I am certain.

**Pull request merged**